### PR TITLE
Define URL for PRs explicitly

### PR DESCRIPTION
**Pull request by kevgo**
As we run into more and more requests to support a larger variety of custom URLs for self-hosted code repos, it starts making sense that our specs define the URLs that `git new-pull-request` uses directly.

This will enable #1019 to remove the hard-coded hostname from our logic to calculate the PR URL moving forward.

### CODE REVIEW

**Comment by charlierudolph** [comment]
Are there some other step definitions we can now delete?

**Comment by kevgo** [comment]
Yes, `cucumber --dry-run -f stepdefs` shows a few unused step defs. None of them are affected by this PR, though. The step def changed here is used in a number of other places as well. 

I will submit a separate PR cleaning up all unused step defs.

File: features/git-town-new-pull-request/bitbucket.feature [hunk 1]
```diff
@@ -11,22 +11,23 @@ Feature: git-new-pull-request when origin is on Bitbucket
...
+      | git@bitbucket.org/Originate/originate.github.com.git              | https://bitbucket.org/Originate/originate.github.com/pull-request/new?dest=Originate%2Foriginate.github.com%3A%3Amain&source=Originate%2Foriginate.github.com%.*%3Afeature |
+      | git@bitbucket.org/Originate/originate.github.com                  | https://bitbucket.org/Originate/originate.github.com/pull-request/new?dest=Originate%2Foriginate.github.com%3A%3Amain&source=Originate%2Foriginate.github.com%.*%3Afeature |
+      | ssh://git@bitbucket.org/Originate/git-town.git                    | https://bitbucket.org/Originate/git-town/pull-request/new?dest=Originate%2Fgit-town%3A%3Amain&source=Originate%2Fgit-town%.*%3Afeature                                     |
+      | ssh://git@bitbucket.org/Originate/git-town                        | https://bitbucket.org/Originate/git-town/pull-request/new?dest=Originate%2Fgit-town%3A%3Amain&source=Originate%2Fgit-town%.*%3Afeature                                     |
```
Comment (charlierudolph) [inline features/git-town-new-pull-request/bitbucket.feature]: This is so verbose! I think we can move a lot of this to unit tests where we get the hostname from the origin url. Then we could get away with only two examples here: one where the host is bitbucket, and the other when the host is an ssh identity: bitbucket-as-account2

Comment (kevgo) [reply features/git-town-new-pull-request/bitbucket.feature]: I have started a spike of a unit test. It ends up even more verbose. We would have to use a [table-driven test](https://github.com/golang/go/wiki/TableDrivenTests), and the table that maps origin to URL would be harder to read than the Gherkin table here. Plus all the boilerplate for the unit test stuff. I believe keeping all examples of PR URLs here in Gherkin is the cleanest, simplest, and most developer-friendly way to test the actual behavior of the tool.

**Comment by kevgo** [comment]
I have implemented the changes we discussed today. Ready for another look!

**Pull request merged**